src/platform/datapath_winuser.c

```diff
@@ -2566,56 +2636,70 @@ QuicDataPathBindingSendFromTo(
     SOCKADDR_INET MappedRemoteAddress = { 0 };
     QuicConvertToMappedV6(RemoteAddress, &MappedRemoteAddress);
 
+    PWSACMSGHDR CMsg;
+    BYTE CtrlBuf[
+        WSA_CMSG_SPACE(sizeof(IN6_PKTINFO)) +   // IP_PKTINFO
+        WSA_CMSG_SPACE(sizeof(INT)) +           // IP_ECN
+#ifdef UDP_SEND_MSG_SIZE
+        WSA_CMSG_SPACE(sizeof(DWORD))    // UDP_SEND_MSG_SIZE
+#endif
+        ];
+
     WSAMSG WSAMhdr;
     WSAMhdr.dwFlags = 0;
     WSAMhdr.name = (LPSOCKADDR)&MappedRemoteAddress;
     WSAMhdr.namelen = sizeof(MappedRemoteAddress);
     WSAMhdr.lpBuffers = SendContext->WsaBuffers;
     WSAMhdr.dwBufferCount = SendContext->WsaBufferCount;
-
-    // TODO - Use SendContext->ECN if not QUIC_ECN_NON_ECT
-
-    PWSACMSGHDR CMsg;
-    BYTE CtrlBuf[WSA_CMSG_SPACE(sizeof(IN6_PKTINFO)) + WSA_CMSG_SPACE(sizeof(*SegmentSize))];
     WSAMhdr.Control.buf = (PCHAR)CtrlBuf;
+    WSAMhdr.Control.len = 0;
 
     if (LocalAddress->si_family == AF_INET) {
-        WSAMhdr.Control.len = WSA_CMSG_SPACE(sizeof(IN_PKTINFO));
-
+        WSAMhdr.Control.len += WSA_CMSG_SPACE(sizeof(IN_PKTINFO));
         CMsg = WSA_CMSG_FIRSTHDR(&WSAMhdr);
         CMsg->cmsg_level = IPPROTO_IP;
         CMsg->cmsg_type = IP_PKTINFO;
         CMsg->cmsg_len = WSA_CMSG_LEN(sizeof(IN_PKTINFO));
-
         PIN_PKTINFO PktInfo = (PIN_PKTINFO)WSA_CMSG_DATA(CMsg);
         PktInfo->ipi_ifindex = LocalAddress->Ipv6.sin6_scope_id;
```

Comment:
This line looks confusing. is Ipv6.sin6_scope_id overloaded for v4 addresses?

Reply:
It's a little hack I use. Ignore it. :)